R/commonAnovaBayesian.R

```diff
@@ -1799,6 +1800,145 @@ BANOVAcomputMatchedInclusion <- function(effectNames, effects.matrix, interactio
+                                               usePooledSE = usePooledSE, dependentName = .BANOVAdependentName,
+                                               subjectName = .BANOVAsubjectName)
+  }
+
+  colnames(summaryStat)[colnames(summaryStat) == dependent] <- "dependent"
+
+  if (options$plotTwoHorizontalAxis != "") {
+    colnames(summaryStat)[colnames(summaryStat) == options$plotTwoHorizontalAxis] <- "plotTwoHorizontalAxis"
+  }
+
+  if (options$plotTwoSeparatePlots != "") {
+    colnames(summaryStat)[colnames(summaryStat) == options$plotTwoSeparatePlots] <- "plotTwoSeparatePlots"
+    subsetPlots <- levels(summaryStat[,"plotTwoSeparatePlots"])
+    nPlots <- length(subsetPlots)
+  } else {
+    nPlots <- 1
+  }
+
+  for (i in seq_len(nPlots)) {
+
+    if (nPlots > 1L) {
+      title <- paste(options$plotTwoSeparatePlots,": ",subsetPlots[i], sep = "")
+    } else {
+      title <- ""
+    }
+    descriptivesPlot <- createJaspPlot(title = title)
+    descriptivesPlotTwoContainer[[title]] <- descriptivesPlot
+
+    descriptivesPlot$height <- 500
+    descriptivesPlot$width <- 500
+
+    if (options$plotTwoSeparatePlots != "") {
+      summaryStatSubset <- subset(summaryStat, summaryStat[,"plotTwoSeparatePlots"] == subsetPlots[i])
+    } else {
+      summaryStatSubset <- summaryStat
+    }
+
+    if (plotErrorBars) {
+      pd <- ggplot2::position_dodge(.2)
+      error <- ggplot2::geom_errorbar(ggplot2::aes(ymin = ciLower, ymax = ciUpper),
+                                      colour = "black", width = .2, position = pd)
+    } else {
+      error <- NULL
+    }
+
+    if (zeroFix) {
+      values <- c(0, summaryStatSubset[,"dependent"],
+                  min(summaryStatSubset[,"dependent"])*1.1, #do we need those extensions?
+                  max(summaryStatSubset[,"dependent"])*1.1)
+    } else {
+      values <- c(summaryStatSubset[,"dependent"],
+                  min(summaryStatSubset[,"dependent"])*1.1,
+                  max(summaryStatSubset[,"dependent"])*1.1)
+    }
+
+    if (plotErrorBars) {
+      ciPos <- c(values, summaryStatSubset$ciLower, summaryStatSubset$ciUpper)
+      yBreaks <- jaspGraphs::getPrettyAxisBreaks(ciPos)
+    } else {
+      yBreaks <- jaspGraphs::getPrettyAxisBreaks(values)
+    }
+    pd2 <- ggplot2::position_dodge2(preserve = "single")
+
+    p <- ggplot2::ggplot(summaryStatSubset, ggplot2::aes(x = plotTwoHorizontalAxis, y = dependent, group = 1)) +
+      ggplot2::geom_hline(yintercept = 0) +
+      ggplot2::geom_bar(stat = "identity", fill = "grey", col = "black", width = .6, position = pd2) +
+      error +
+      ggplot2::labs(y = yLabel, x = options[["plotTwoHorizontalAxis"]]) +
+      ggplot2::scale_y_continuous(breaks = yBreaks, limits = range(yBreaks), oob = scales::rescale_none) +
+      ggplot2::scale_x_discrete(breaks = jaspGraphs::getPrettyAxisBreaks(summaryStatSubset[,"plotTwoHorizontalAxis"])) +
+      jaspGraphs::geom_rangeframe(sides = "l") +
+      jaspGraphs::themeJaspRaw()
```

Comment:
I honestly don't know how much this would be used in other modules as it is a really basic bar plot set up of couple of lines of code. Maybe there are more sophisiticated bar visualizations, but I can't think of any right now. Maybe a topic for another PR focusing on reducing code duplication in general?